src/hooks/useUploadReferral.ts

```diff
@@ -18,10 +21,28 @@ const useUploadReferral = () => {
     const handleFileChange = (e: React.ChangeEvent<HTMLInputElement>) => {
         const file = e.target.files?.item(0);
         if (file && isValidFile(file)) {
-            setFile(file);
+            uploadReferralSheet(usersMock[0].recipientUserId, file).then(() => setFile(file))
         }
     };
 
+    const uploadReferralSheet = (recipientUserId: string, file: File) => {
+        return RecipientUserService.getReferralSheetUploadUrl(recipientUserId)
+            .then((referralSheetUploadUrl) => uploadFileToReferralSheetUploadUrl(referralSheetUploadUrl, file))
+            .then((referralSheetUrl)=> RecipientUserService.updateReferralSheetUrl(recipientUserId, referralSheetUrl))
+    }
+
+    const uploadFileToReferralSheetUploadUrl = (referralSheetUploadUrl: string, file: File) => {
+        return axios.put(referralSheetUploadUrl, file)
+            .then(()=> getReferralSheetUrlFromReferralSheetUploadUrl(referralSheetUploadUrl))
+    }
+
+    const getReferralSheetUrlFromReferralSheetUploadUrl = (referralSheetUploadUrl: string) => {
+        return referralSheetUploadUrl
+            .replace("https://", "")
+            .split("/")[2]
+            .split("?")[0];
```

Comment:
This approach seems to be brittle. Do we always use `https` protocol? If the url has always the same structure I think it would be better of using regex and appending the right groups.

Reply:
I know, this is horrible, I was thinking that RecipientUserService.getReferralSheetUploadUrl(recipientUserId) should return the relative path of the s3 /referral-sheets/referral_sheet_{userId}_{date} as well as the ReferralSheetUploadUrl.

Because the client shouldn't know about this path format that can change in the backend at anytime, it would be better if the backend saves the file however it pleases and then return this path to the client

What do you think? I can make this changes in the backend